src/test/java/io/vertx/tests/validation/transformer/DeepObjectTransformerTest.java

```
@@ -0,0 +1,67 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class DeepObjectTransformerTest implements SchemaSupport {
+
+  private static final Parameter DEEP_OBJECT_PARAM = mockParameter(NAME, QUERY, DEEP_OBJECT, true, OBJECT_SCHEMA);
+  private static final Parameter DEEP_OBJECT_ARRAY_PARAM = mockParameter(NAME, QUERY, DEEP_OBJECT, true, ARRAY_SCHEMA);
+
+  private static final DeepObjectTransformer TRANSFORMER = new DeepObjectTransformer();
+
+  private static Stream<Arguments> provideValidObjectValues() {
+    String complexExplodedRaw = "dummy[role]=admin&dummy[firstName]=Alex";
+    String complexExplodedRawWithExtras = "something=nothing&dummy[role]=admin&dummy[firstName]=Alex";
+    String complexExplodedRawWithAnotherParameter =
+        "dummy[role]=admin&dummy[firstName]=Alex&silly[role]=user&silly[firstName]=John";
+    JsonObject expected = new JsonObject().put("role", "admin").put("firstName", "Alex");
+
+    return Stream.of(
+        Arguments.of("empty", DEEP_OBJECT_PARAM, "", EMPTY_JSON_OBJECT),
+        Arguments.of(complexExplodedRaw + " (exploded)", DEEP_OBJECT_PARAM, complexExplodedRaw, expected),
+        Arguments.of(complexExplodedRawWithExtras + " (exploded)", DEEP_OBJECT_PARAM, complexExplodedRawWithExtras,
+            expected),
+        Arguments.of(complexExplodedRawWithAnotherParameter + " (exploded)", DEEP_OBJECT_PARAM,
+            complexExplodedRawWithAnotherParameter, expected));
+  }
+
+  @ParameterizedTest(name = "{index} Transform \"Query\" parameter of style \"deepObject\" with object value: {0}")
+  @MethodSource("provideValidObjectValues")
+  void testTransformObjectValid(String scenario, Parameter parameter, String rawValue, Object expectedValue) {
+    assertThat(TRANSFORMER.transformObject(parameter, rawValue)).isEqualTo(expectedValue);
+  }
+
+  @Test
+  void testInvalidTransformation() {
```

Where is the test that shows what happens when the parameter is neither an object nor an array?

---

I've added primitive tests

---

I'm not sure if this is the correct behavior. The documentation says it is n/a [1]. So maybe we should also throw an unsupported transformation error, what do you think?

[1] https://swagger.io/docs/specification/v3_0/serialization/#query-parameters

---

I think N/A means we should throw an error. I had initially left the base implementation as with the other transformers, but I have changed it now.